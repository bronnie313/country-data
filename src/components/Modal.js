import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import './styles/Modal.css';

const Modal = () => {
  const { countries, selectedCountryId } = useSelector((store) => store.country);
  const selectedCountry = countries.find((country) => country.id === selectedCountryId);

  if (!selectedCountry) {
    return null;
  }

  const {
    id, name, population, area, flags, region, continents, capital, maps,
  } = selectedCountry;
  const [, flagURLW320] = flags;
  const [continent] = continents;
  const [capitalCity] = capital;
  const { common } = name;
  const { googleMaps } = maps;

  return (
    <aside>
      <div className="modal">
        <SelectedModal
          country={common}
          capital={capitalCity}
          continent={continent}
          area={area}
          population={population}
          flags={flagURLW320}
          region={region}
          googleMaps={googleMaps}
          id={id}
        />
      </div>
    </aside>
  );
};

export const SelectedModal = (props) => {
  const {
    country, population, area, flags, region, continent, capital, googleMaps,
  } = props;

  return (
    <div className="modal-div">
      <img className="modal-img" src={flags} alt="Country Flag" />
      <ul className="modal-ul">
        <li>
          <h2>Country:</h2>
          <h2>{country}</h2>
        </li>
        <li>
          <h3>Capital:</h3>
          <h3>{capital}</h3>
        </li>
        <li>
          <h4>Population: </h4>
          <h4>
            {population}
            {' '}
            <i>people</i>
          </h4>
        </li>
        <li>
          <h4>Area: </h4>
          <h4>{area}</h4>
        </li>
        <li>
          <h4>Region: </h4>
          <h4>{region}</h4>
        </li>
        <li>
          <h4>Continent:</h4>
          <h4>
            {' '}
            {continent}
          </h4>
        </li>
        <li><a href={googleMaps}>googleMaps</a></li>
      </ul>
    </div>
  );
};

SelectedModal.propTypes = {
  flags: PropTypes.string.isRequired,
  population: PropTypes.number.isRequired,
  area: PropTypes.number.isRequired,
  capital: PropTypes.string.isRequired,
  region: PropTypes.string.isRequired,
  continent: PropTypes.string.isRequired,
  googleMaps: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
};

export default Modal;
