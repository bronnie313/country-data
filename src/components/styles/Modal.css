aside {
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 4.3rem;
  bottom: 0;
  left: 0;
  right: 0;
  background: #fd5294;
  color: white;
}

.image {
  width: 50%;
}

.modal-ul {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.modal-ul > li {
  background: #ec4c8a;
  width: 100%;
  padding: 1rem 0;
}

.modal-ul > li:nth-child(2n) {
  background: transparent;
}
