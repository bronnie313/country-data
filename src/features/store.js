import { configureStore } from '@reduxjs/toolkit';
import modalReducer from './modal/modalSlice';
import countryReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    modal: modalReducer,
    country: countryReducer,
  },
});

export default store;
