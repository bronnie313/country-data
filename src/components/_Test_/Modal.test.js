import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../features/store';
import Modal from '../Modal';

describe('home component', () => {
  it('testing if page matches', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Modal />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
