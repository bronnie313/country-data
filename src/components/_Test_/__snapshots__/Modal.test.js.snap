// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component testing if page matches 1`] = `null`;
